### PR TITLE
Update plot

### PR DESCRIPTION
**Pull request by alexmaragko**
Update plot function in base.py and run_pahfit script.
- Residuals panel added
- Errorbars with uncertainties added
- Tickmarks updated
M101_Nucleus_irs

### CODE REVIEW

**Review by karllark — COMMENTED** [review]
Looks great.

I do wonder if the having the residual plot being a bit smaller (maybe height ratios of 3 to 1?) might be better.  

Also setting the default y-scale for the residuals to something other than min/max would be really nice.  Right now the very large noise at the edge of the wavelength range makes it hard to see the majority of the residuals.  Maybe compute the standard deviation of the residuals and then base the y-axis range to be +/- n*stddev where n would be 2 or 3?

**Comment by alexmaragko** [comment]
> Looks great.
> 
> I do wonder if the having the residual plot being a bit smaller (maybe height ratios of 3 to 1?) might be better.
> 
> Also setting the default y-scale for the residuals to something other than min/max would be really nice. Right now the very large noise at the edge of the wavelength range makes it hard to see the majority of the residuals. Maybe compute the standard deviation of the residuals and then base the y-axis range to be +/- n*stddev where n would be 2 or 3?

That's a good suggestion. I'll work on those first chance.

**Comment by alexmaragko** [comment]
M101_Nucleus_irs
- Height ratio set to 3:1
- ylim +/- 2*std of the residuals

**Comment by ThomasSYLai** [comment]
@alexmaragko Thanks for adding this function! This can be very helpful. I have a small suggestion here. Perhaps adding a horizontal line with light color at residual=0 will help us better spot places with larger discrepancies.

**Comment by alexmaragko** [comment]
Sure thing! By the way any idea why the codestyle test is failing? It states "pahfit --count --max-line-length=100" but I'm not sure where this is happening.

**Comment by karllark** [comment]
> By the way any idea why the codestyle test is failing? It states "pahfit --count --max-line-length=100" but I'm not sure where this is happening.

The four lines above that state what lines have codestyle errors.

```
pahfit/base.py:271:28: E261 at least two spaces before inline comment
pahfit/base.py:313:29: E226 missing whitespace around arithmetic operator
pahfit/base.py:320:23: E226 missing whitespace around arithmetic operator
pahfit/base.py:320:30: E226 missing whitespace around arithmetic operator
```

**Comment by karllark** [comment]
You can lookup the E??? values to learn the details.

**Comment by alexmaragko** [comment]
M101_Nucleus_irs

File: pahfit/base.py [hunk 1]
```diff
@@ -287,8 +300,24 @@ def plot(self, ax, x, y, model):
...
+        # residuals
+        res = (y - model(x)) / x
+        std = np.std(res)
+        ax = axs[1]
+        ax.minorticks_on()
+        ax.tick_params(which='both', right='on', top='on', direction='in')
+        ax.axhline(0, linestyle='--', color='gray', zorder=0)
+        ax.plot(x, res, color='k', zorder=1)
+
+        ax.set_ylim(-2 * std, 2 * std)
```
Comment (karllark) [inline pahfit/base.py:316]: It would be great if you could put in the scale factor (here 2) as a keyword input to this function (e.g., `scalefac_resid=2` in the command line) so that it can be changed by users.

File: pahfit/scripts/run_pahfit.py [hunk 2]
```diff
@@ -137,18 +137,19 @@ def main():
...
 
-    pmodel.plot(ax, obs_x, obs_y, obs_fit)
-
-    ax.set_yscale("linear")
-    ax.set_xscale("log")
+    pmodel.plot(axs, obs_x, obs_y, obs_unc.value, obs_fit)
 
     # use the whitespace better
-    fig.tight_layout()
+    fig.subplots_adjust(hspace=0)
```
Comment (karllark) [inline pahfit/scripts/run_pahfit.py:152]: Why the change from tight_layout to subplots_adjust?  Not a problem, I'm curious and want to learn.

**Review by karllark — CHANGES_REQUESTED** [review]
Looks awesome.  Very minor change request.

File: pahfit/base.py (hunk 1 above)
Comment (alexmaragko) [reply pahfit/base.py:316]: Sure! I'll try to do it sometime today or tomorrow and also update plot_pahfit.py script

File: pahfit/scripts/run_pahfit.py (hunk 2 above)
Comment (alexmaragko) [reply pahfit/scripts/run_pahfit.py:152]: I had fig.subplots_adjust(hspace=0) above the tight_layout line without removing the tiight_layout() but then tight_layout overrode hspace=0 and made the panels with an offset.

**Review by karllark — APPROVED** [review]
Looks awesome!  Thanks.
